public/app/features/alerting/unified/components/GrafanaAlertmanagerDeliveryWarning.test.tsx

```diff
@@ -1,39 +1,20 @@
 import { render, screen, waitFor } from '@testing-library/react';
-import { setupServer } from 'msw/node';
 import React from 'react';
 import { Provider } from 'react-redux';
 
-import { setBackendSrv } from '@grafana/runtime';
-import { backendSrv } from 'app/core/services/backend_srv';
+import { setupMswServer } from 'app/features/alerting/unified/mockApi';
+import { setAlertmanagerChoices } from 'app/features/alerting/unified/mocks/server/configure';
 import { configureStore } from 'app/store/configureStore';
 
 import { AlertmanagerChoice } from '../../../../plugins/datasource/alertmanager/types';
-import { mockAlertmanagerChoiceResponse } from '../mocks/alertmanagerApi';
 import { GRAFANA_RULES_SOURCE_NAME } from '../utils/datasource';
 
 import { GrafanaAlertmanagerDeliveryWarning } from './GrafanaAlertmanagerDeliveryWarning';
+setupMswServer();
 
 describe('GrafanaAlertmanagerDeliveryWarning', () => {
```

Comment:
I think it would be nice to have the ability to setup handlers per tests file and per single test
In many scenarios you'd like to setup necessary endpoints only once and then setup a different handler per each test for the thing you're actually testing

Reply:
That's the pattern that I'm aiming to solve with the config methods - so the handlers behave in a predefined way (by us) across the tests, and when you want to tweak it, you create a new method that modifies the behaviour. 

This would be done for things that we can't handle by just adding another item in an array of responses (e.g., if we want to test lots of different alert group/rule configs, we just have the relevant APIs return different examples, rather than making that endpoint start returning an entirely different thing. We can then use test fixtures across the mock API code and our tests, so we'd say (pseudocode)

```ts
render(<SomeComponent ruleId={MOCK_RULE_ID_GRAFANA_MANAGED} />)

// The API handlers then also do stuff with `MOCK_RULE_ID_GRAFANA_MANAGED`
//...

expect('some thing').toBeInTheDocument()
```